stdlib/typing.pyi

```diff
@@ -471,7 +472,7 @@ class MutableMapping(Mapping[_KT, _VT], Generic[_KT, _VT]):
     # known to be a Mapping with unknown type parameters, which is closer
     # to the behavior we want. See mypy issue  #1430.
     @overload
-    def update(self, __m: Mapping[_KT, _VT], **kwargs: _VT) -> None: ...
+    def update(self, __m: SupportsKeyAndGetItem[_KT, _VT], **kwargs: _VT) -> None: ...
```

Comment:
Notes for reviewers:
- Is it ok to change this? If not, I can instead add `# type: ignore` to `dict.update()`.
- The big comment above needs updating, but how should I update it? I'm not sure what parts of it still apply.